### ISSUE TITLE
Normalization with constant time series dimensions

### ISSUE THREAD

**Issue by jircio** [maxar/IncDTW#2]
Hi Maximilian,
we are encountering an issue with the function norm() and with the functions like rundtw() that call for normalization. If a multivariate time series have a constant dimension or the univariate time series is constant, the process of normalization does not finish correctly (it returns NAN values).
the reason for this error is that the possibility of dividing by 0 is not being considered. For example for z normalization:
```
// [[Rcpp::export]]
NumericVector cpp_znorm(NumericVector x,
                        Rcpp::Nullable< Rcpp::NumericVector > mu_in = R_NilValue,
                        Rcpp::Nullable< Rcpp::NumericVector > sd_in = R_NilValue)
 {
   
   int N = x.size();
   NumericVector y(N);
   double mu;
   double sd;
   
   if (mu_in.isNull() || sd_in.isNull()) {
      mu = 0;
      sd = 0;
      
      for(int i = 0; i < N; i++ ){
         mu += x[i];
      }
      mu /= N;
   
      for(int i = 0; i < N; i++ ){
         y[i] = (x[i] - mu);
         sd += y[i] * y[i];
      }
      sd /= (N-1);
      sd = sqrt(sd);
   
      for(int i = 0; i < N; i++ ){
         y[i] /= sd;      // **if sd is 0 it returns a NAN.**
      }
      
   }else{
      
      mu = Rcpp::as< Rcpp::NumericVector >(mu_in)[0];
      sd = Rcpp::as< Rcpp::NumericVector >(sd_in)[0];
      for(int i = 0; i < N; i++ ){
         y[i] = (x[i] - mu)/sd;
      }
   }
   
   return y;
}
```

It would be great if you could have a look into this.
Cheers

**Comment by maxar** [comment]
Thanks! I'll take a look tomorrow and let you know.
Cheers

**Comment by maxar** [comment]
Should be solved with new version 1.1.2.01, currently only here in my repo on Github.

Now there is a check if the sd is smaller than the parameter threshold (default = 1e-5). If the sd is smaller than the threshold, then no scaling is performed, only shifting. Analogous it's implemented for min-max normalization. Also for rundtw().

Please let me know if this works for you?

**Comment by jircio** [comment]
It's working properly. Thank you very much!
